### DISCUSSION TITLE
IntelliJ Copilot visual bug

### DISCUSSION THREAD

**Discussion by NedimBadzak** [community/community#8965] [Copilot Conversations] [answered]
First off, let me start by saying I love your Github Copilot product. I was using it normally and for smaller code snippets most often. Today, after typing the function definition, I was presented with an already complete function which is quite a few rows long, and instead of displaying it in the "text portion" of IntelliJ IDEA, it was actually displayed on top of all the menus (as you can see in the screenshot below).

![Screenshot from 2021-12-16 19-01-30](https://user-images.githubusercontent.com/23142003/146424811-f09e4dcc-93ff-45c1-96e2-cc667b09cde3.png)

I think that this isn't a deal breaker or anything like that, but a rather small bug that could easily be fixed, but contributes to the future user experience a lot.

System:
Ubuntu 21.10
IntelliJ IDEA 2021.3 (Ultimate Edition) Build #IU-213.5744.223, built on November 27, 2021
Github Copilot plugin 1.1.4.220
 
Best regards :)

**Comment by jansorg** [comment]
Thanks for your feedback!
That's very odd. Are you consistently seeing this kind of text rendering?

GitHub Copilot is using functionality of the IDE's SDK which restricts the rendering to the current editor. It's supposed to scroll with the editor content and not overlay everything else.

**Comment by NedimBadzak** [comment]
Thanks for the reply!

Well, I wouldn't say that it is consistent, but even if doesn't overlay everything else at first (which it does 4/10 times), it does every time when I try to scroll a bit. 
I also noticed that, it is only shown in the empty parts of underlying parts of the editor, so if you have some text, that part of the suggestion is not rendered over it, but rather under it or not at all.
Also when I scroll, the suggestion scrolls over the editor elements, so it's not statically over them but rather keeps being updated over them.

**Comment by jansorg** [comment]
@NedimBadzak Are you still seeing these rendering issues?
If that's still the case, are you perhaps using a custom JVM setup for your IDE? Please provide the info shown at `Help > About`.

**Reply by NedimBadzak** [reply]
Yes, I am still seeing these rendering issues, and I don't think that I am using a custom JVM setup.
Here is the `Help > About` part:
```
IntelliJ IDEA 2021.3 (Ultimate Edition)
Build #IU-213.5744.223, built on November 27, 2021
Licensed to Nedim Badzak
Subscription is active until June 5, 2022.
For educational use only.
Runtime version: 11.0.13+7-b1751.19 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.13.0-28-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 1730M
Cores: 16
Non-Bundled Plugins:
    krasa.CpuUsageIndicator (1.14)
    de.ax.powermode (100.001)
    com.wakatime.intellij.plugin (13.1.7)
    org.sonarlint.idea (6.3.1.40498)
    org.jetbrains.kotlin (213-1.6.10-release-923-IJ5744.223)
    com.github.vljubovic.javaFxHelper (1.0.8)
    com.github.copilot (1.1.4.220)
    PMDPlugin (1.8.21)
    Key Promoter X (2021.3)
    Dart (213.5744.122)
    io.flutter (63.2.4)
    com.nbadal.ktlint (0.8.2)
    izhangzhihao.rainbow.brackets (6.21)

Kotlin: 213-1.6.10-release-923-IJ5744.223
Current Desktop: ubuntu:GNOME

```
I tried ruling out certain plugins, but none of them seem to be the culprit. 

Hope this output helps.

**Reply by jansorg** [reply]
Thanks @NedimBadzak! You're running the default JetBrains JVM build, which is good.
Unfortunately, we're unable to reproduce this on Linux and this is the only report we've received about such a glitch. As far as I understand the code, this kind of graphical glitch is outside of our plugin's control. It seems more like an issue with the JVM or a combination of JVM and graphics drivers, but that's only a random guess.

For context, could you add which graphics drivers your're using (e.g. Intel, NVIDIA, AMD) if you're using X11 or wayland?

**Reply by RikEnde** [reply]
I see the same issue on a mac

IntelliJ IDEA 2021.3.2 (Ultimate Edition)
Build #IU-213.6777.52, built on January 27, 2022
Licensed to <my full name>
Subscription is active until July 30, 2022.
Runtime version: 11.0.13+7-b1751.25 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 11.6.3
GC: G1 Young Generation, G1 Old Generation
Memory: 750M
Cores: 16
Registry:
    debugger.watches.in.variables=false

Non-Bundled Plugins:
    org.intellij.plugins.hcl (0.7.14)
    name.kropp.intellij.makefile (213.5744.224)
    in.xiv.cs (0.1)
    com.github.copilot (1.1.6.322)
    com.euphoricity.gitignore (0.3)
    appland.appmap (0.7.6)
    mobi.hsz.idea.gitignore (4.3.0)
    CheckStyle-IDEA (5.57.2)
    CFML Support (213.5744.223)
    MavenRunHelper (4.18.212.000.3)
    com.microsoft.tooling.msservices.intellij.azure (3.61.1-2021.3)

Kotlin: 213-1.5.10-release-949-IJ6777.52

<img width="839" alt="Screen Shot 2022-02-10 at 4 57 12 PM" src="https://user-images.githubusercontent.com/6178351/153603669-106c2bbd-cf6c-4932-b0fb-c792a39af3f2.png">

**Comment by jansorg** [comment]
@NedimBadzak After another user reported this, I was able to reproduce this. For me, this is happening if a very long suggestion is rendered in the editor and if I then interact with tool windows, e.g. resizing a tool window below the editor. Thanks!

**Comment by jansorg** [comment, marked as answer]
@NedimBadzak @RikEnde We published a new release today, which should fix the bug you've seen. The release (v 1.1.12) should be available later today or tomorrow after the review by JetBrains.
